### PR TITLE
ebitenutil: added NewImageFromReader(io.Reader)

### PR DESCRIPTION
**Pull request by TheMightyGit**
It's additionally useful if ebitenutil also supports an `io.Reader` method `NewImageFromReader(reader io.Reader)`.

### CODE REVIEW

File: ebitenutil/loadimage_go_1_16.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
+// Copyright 2014 Hajime Hoshi
```
Comment (hajimehoshi) [inline ebitenutil/loadimage_go_1_16.go:1]: Copyright 2021 The Ebiten Authors

File: ebitenutil/loadimage_go_1_16.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build (darwin || freebsd || js || linux || windows) && !android && !ios && go1.16
+// +build darwin freebsd js linux windows
+// +build !android
+// +build !ios
```
Comment (hajimehoshi) [inline ebitenutil/loadimage_go_1_16.go:18]: Why are the mobiles opted out? I think this should work in every environment.

File: ebitenutil/loadimage_go_1_16.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+// NewImageFromFS loads the file from a fs.FS filesystem with path and returns ebiten.Image and image.Image.
+//
+// go1.16+ only
+//
+// Image decoders must be imported when using NewImageFromFS. For example,
+// if you want to load a PNG image, you'd need to add `_ "image/png"` to the import section.
+//
+// This is useful if you've ebedded a filesystem with embed.FS.
+func NewImageFromFS(filesystem fs.FS, path string) (*ebiten.Image, image.Image, error) {
```
Comment (hajimehoshi) [inline ebitenutil/loadimage_go_1_16.go:38]: Hmm, while `NewImageFromReader` is fine, I tend not to want to have this.

```
f, err := filesystem.Open(path)
if err != nil {
    return err
}
defer f.Close()
img, _, err := ebitenutil.NewImageFromReader(f)
if err != nil {
    return err
}
// Use img
```

Is this so burden?

File: ebitenutil/loadimage_go_1_16.go (hunk 2 above)
Comment (TheMightyGit) [reply ebitenutil/loadimage_go_1_16.go]: I've added ios and android - but should I instead remove all os tags?

File: ebitenutil/loadimage_go_1_16.go (hunk 3 above)
Comment (TheMightyGit) [reply ebitenutil/loadimage_go_1_16.go:36]: I agree it has less value than `NewImageFromReader` (which only has a bad alternative). I'll remove it from this PR.

File: ebitenutil/loadimage_go_1_16.go (hunk 2 above)
Comment (TheMightyGit) [reply ebitenutil/loadimage_go_1_16.go]: this file is removed now.

Comment (hajimehoshi) [reply ebitenutil/loadimage_go_1_16.go]: Ah yeah I intended to remove all the tags first. I acknowledged this file was removed.

**Review by hajimehoshi — APPROVED** [review]
LGTM, but is this still a draft?

**Pull request merged**